From 32a6b7830af50347b9ff7c306809c7e9d8327a52 Mon Sep 17 00:00:00 2001
From: He Junyan <junyan.he@intel.com>
Date: Mon, 11 Sep 2023 23:10:00 +0800
Subject: [PATCH 78/78] glcolorconvert: Calculate correct swizzles for Y410,
 Y412 formats

The GST_VIDEO_FORMAT_Y410, GST_VIDEO_FORMAT_Y412_LE and GST_VIDEO_FORMAT_Y412_BE
formats in fact are packed formats, which have just 1 plane. But we have special
setting for them rather than using get_single_planar_format_gl_swizzle_order().

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5314>
---
 gst-libs/gst/gl/gstglformat.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglformat.c b/gst-libs/gst/gl/gstglformat.c
index 6112a0ba..240a8e82 100644
--- a/gst-libs/gst/gl/gstglformat.c
+++ b/gst-libs/gst/gl/gstglformat.c
@@ -511,7 +511,10 @@ gst_gl_video_format_swizzle (GstVideoFormat video_format, int *swizzle)
 {
   const GstVideoFormatInfo *finfo = gst_video_format_get_info (video_format);
 
-  if (finfo->n_planes == 1) {
+  if (finfo->n_planes == 1 &&
+      (finfo->flags & GST_VIDEO_FORMAT_FLAG_RGB ||
+          video_format == GST_VIDEO_FORMAT_AYUV ||
+          video_format == GST_VIDEO_FORMAT_VUYA)) {
     get_single_planar_format_gl_swizzle_order (video_format, swizzle);
     return TRUE;
   }
-- 
2.40.1

